### PR TITLE
add savegame conflict resolution api

### LINKED ISSUE #8: Expose API to resolve conflicting saves
See https://developer.apple.com/documentation/gamekit/saving_the_player_s_game_data_to_an_icloud_account/#4277691

Thanks!

### PR DESCRIPTION
**Pull request by extrawurst**
closes #8

### CODE REVIEW

**Comment by Seldom-SE** [comment]
I'll need to refactor a bit on my end to test this. Since `ConflictingSaveGames` contains a list of `IosGCSaveGame`s instead of the saves' bytes, that means I'll need to use `load_game` to get the bytes. And that means when I receive a `LoadGame`, I don't know whether I received something I was trying to load to the world, or something that has conflicts I need to resolve. It would be cool if the crate kept track of state like that, though it looks like this crate is meant to be a light wrapper, which is understandable. I can work around it, I'll get back to you tomorrow probably. Thanks!

File: crate/src/plugin.rs [hunk 1]
```diff
@@ -2,35 +2,39 @@ use bevy_app::prelude::*;
...
     /// Triggered by calls to [`achievements_reset`][crate::achievements_reset]
-    AchievementsReset(IosGCAchievementsResetResponse),
+    AchievementsReset((i64, IosGCAchievementsResetResponse)),
     /// Triggered by calls to [`leaderboards_score`][crate::leaderboards_score]
-    LeaderboardScoreSubmitted(IosGCScoreSubmitResponse),
+    LeaderboardScoreSubmitted((i64, IosGCScoreSubmitResponse)),
     /// Triggered by calls to [`fetch_signature`][crate::fetch_signature]
-    ItemsForSignatureVerification(IosGCFetchItemsForSignatureVerificationResponse),
+    ItemsForSignatureVerification((i64, IosGCFetchItemsForSignatureVerificationResponse)),
+    /// Triggered by calls to [`fetch_save_games`][crate::fetch_save_games] or [`save_game`][crate::save_game]
```
Comment (Seldom-SE) [inline crate/src/plugin.rs:34]: I'm still on `fe5ac99`, and I'm receiving `ConflictingSaveGames` 5 seconds after I call `save_game` and it isn't mutually exclusive with `SavedGame`. Maybe that's expected, but it seemed weird, so I thought I'd let you know

**Comment by Seldom-SE** [comment]
Also on `fe5ac99`, when I call `load_game` on the saves I get from `ConflictingSaves`, I get a `LoadGame(Unknown)` event, so I can't actually get the binary save data I need for `resolve_conflicting_games`. Am I requesting the data wrong?

I have some other tasks I need to prioritize, so I'm not sure when I'll be back to this, sorry. Thanks for implementing this feature!

**Comment by extrawurst** [comment]
Would be great if you can upgrade to latest `212ec9c` (it does not force you to use the observer API). My problem with testing this is: that my sandbox account seems to be rate limited or in any other way borked and with real apple id i don't have two apple devices :D and i am not using my real apple id on a simulator. long story short: i have a hard time reproducing conflicts for myself to test this.
what i noted for sure is that simulator builds behave very differently to store builds. so maybe you can try if that makes a difference?

**Comment by extrawurst** [comment]
Btw. according to https://developer.apple.com/documentation/gamekit/gklocalplayer/fetchsavedgames(completionhandler:) you can also pro-actively know what savegames conflict based on duplicate filenames in the response to `fetchsavedgames`

File: crate/src/plugin.rs (hunk 1 above)
Comment (extrawurst) [reply crate/src/plugin.rs:34]: > receiving ConflictingSaveGames 5 seconds after I call

there is really no waiting involved in the code triggered on native side to report to us. but i observe a whole lot of slow behaviour with gamecenter in debug simulator builds. sometimes fetchsavegames just never returns for me

>  it isn't mutually exclusive with `SavedGame`

what does that mean?

**Pull request merged**